### DISCUSSION TITLE
Normalize log formats across projects

### DISCUSSION THREAD

**Discussion by nynymike** [JanssenProject/jans#896] [Ideas]
Log messages come in different formats; this makes it hard to parse them correctly.

For example, we see logs like:

```
 example from scim
07-02 15:17:16.714 INFO  gluu.service.logger.LoggerService LoggerService.java:205- Updated log level of '21' loggers to INFO

# example from oxtrust
INFO - webdav - 2022-02-07 15:15:24,402 - Sync files from http://jackrabbit:8080/repository/default/opt/gluu/jetty/identity/custom

# example from oxtrust 2:
2022-02-07 15:11:05.626:INFO :oejs.AbstractConnector:main: Started ServerConnector@15c4953d{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}

# example from oxtrust 3
2022-02-07 15:11:01,589 INFO  [main] [org.gluu.jsf2.customization.FacesLocalizationConfigPopulator] (FacesLocalizationConfigPopulator.java:95) - Adding languages '[en, fr, ru]' from dir folder: /opt/gluu/jetty/identity/webapps/identity/WEB-INF/classes/
```

**Comment by moabu** [comment]
lets agree on a style for our logs that basically everything should follow

whether its java, python or whatever

@ossdhaval  following this discussion we will need a doc to be added adding the default style of log output we agree on so please add that once we close this thread

**Comment by ossdhaval** [comment]
In my experience, log message structure as we see in the last sample above, to be most useful and easy to parse.

`# example from oxtrust 3
2022-02-07 15:11:01,589 INFO  [main] [org.gluu.jsf2.customization.FacesLocalizationConfigPopulator] (FacesLocalizationConfigPopulator.java:95) - Adding languages '[en, fr, ru]' from dir folder: /opt/gluu/jetty/identity/webapps/identity/WEB-INF/classes/
`
We also need to see recommendations on logging format from cloud logging and monitoring tools.

**Reply by moabu** [reply]
We want to unify the log format. We need to agree on the the global log structure and all logging including any python or cloud logging would need to follow it. Its not just about the format. I think we need to start thinking about **structured logging**.

**Reply by ossdhaval** [reply]
yes, JSON structured logging would be a natural choice as Janssen support cloud-native deployments. And tooling for same is widely available and seamlessly integrated in our current design.

**Comment by ossdhaval** [comment]
I am closing this discussion as it'll tracked as an issue.
